mapie/time_series_regression.py

```diff
@@ -0,0 +1,272 @@
+                method="lower",
+            )
+            betas_0[:, ind_alpha] = betas[
+                np.argmin(one_alpha_beta - beta, axis=0)
+            ]
+
+        return betas_0
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Compared to the method ``fit`` of ``MapieRegressor``, the ``fit``
+        method of ``MapieTimeSeriesRegressor`` computes the
+        ``conformity_scores_`` with relative values.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+        """
+        self = super().fit(X=X, y=y, sample_weight=sample_weight)
+        self.conformity_scores_ = self._relative_conformity_scores(X, y)
+        return self
+
+    def partial_fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Update the ``conformity_scores_`` attribute when new data with known
+        labels are available.
+        Note: Don't use ``partial_fit`` with samples of the training set.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples_test, n_features)
+            Input data.
+
+        y : ArrayLike of shape (n_samples_test,)
+            Input labels.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+
+        Raises
+        ------
+        ValueError
+            If the length of y is greater than the length of the training set.
+        """
+        X = cast(NDArray, X)
+        y = cast(NDArray, y)
+        n = len(self.conformity_scores_)
+        if len(X) > n:
+            raise ValueError(
+                "You try to update more residuals than there are!"
+            )
+        new_conformity_scores_ = self._relative_conformity_scores(X, y)
+        self.conformity_scores_ = np.roll(
+            self.conformity_scores_, -len(new_conformity_scores_)
+        )
+        self.conformity_scores_[
+            -len(new_conformity_scores_):
+        ] = new_conformity_scores_
+        return self
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        alpha: Optional[Union[float, Iterable[float]]] = None,
+        optimize_beta: bool = True,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        Correspond to 'Conformal prediction for dynamic time-series'.
+
+        Parameters
+        ----------
+        optimize_beta: bool
+            Whether to optimize the PIs' width or not.
+
+        """
+
+        # Checks
+        check_is_fitted(self, self.fit_attributes)
+        self._check_ensemble(ensemble)
+        alpha = cast(Optional[NDArray], check_alpha(alpha))
+        y_pred = self.single_estimator_.predict(X)
+        n = len(self.conformity_scores_)
+
+        if alpha is None:
+            return np.array(y_pred)
+        else:
```

Comment:
`else:` is not needed (see `MapieRegressor`)

Reply:
Done